.github/workflows/commit-validation.yml

```
@@ -0,0 +1,105 @@
+          
+          if echo "$PR_TITLE" | grep -qE "^(feat|fix|docs|style|refactor|perf|test|build|ci|chore|revert)(\(.+\))?: .+"; then
+            echo "✅ Valid PR title: $PR_TITLE"
+          else
+            echo "⚠️ PR title should follow conventional commits format"
+            echo "Current: $PR_TITLE"
+            echo "Expected: type(scope): description"
+            exit 1
+          fi
+
+      - name: Validate PR description
+        run: |
```

actionlint (1.7.9)

"github.event.pull_request.body" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)